caps/repo.go

```diff
@@ -0,0 +1,164 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package caps
+
+import (
+	"fmt"
+	"sort"
+	"sync"
+)
+
+// Repository stores all known snappy capabilities and types
+type Repository struct {
+	// Protects the internals from concurrent access.
+	// If contention gets high, switch to a RWMutex
```

Comment:
You're right. I meant it more as a “doing the simplest thing that works” comment, an appeal to not get ahead of ourselves. I should probably just have made it as a review comment, not a code comment.